docs/eventing/samples/writing-event-source-easy-way/README.md

```diff
@@ -0,0 +1,305 @@
+
+[ContainerSource](../../../eventing/sources/README.md#meta-sources) is an easy way to turn any dispatcher container into an Event Source.
+Similarly, another option is using [SinkBinding](../../../eventing/sources/README.md#meta-sources)
+which provides a framework for injecting environment variables into any Kubernetes resource which has a `spec.template` that looks like a Pod (aka PodSpecable).
+
+Code for this tutorial is available at [Github](https://github.com/aliok/node-knative-heartbeat-source).
+
+# Bootstrapping
+
+Create the project and add the dependencies:
+```bash
+npm init
+npm install cloudevents-sdk --save
+```
+
+## Making use of ContainerSource
+
+`ContainerSource` and `SinkBinding` both work by injecting environment variables to an application.
+Injected environment variables at minimum contain the URL of a sink that will receive events.
+
+Following example emits an event to the sink every 1000 milliseconds.
+The sink URL to post the events will be made available to the application via the `K_SINK` environment variable by `ContainerSource`.
+
+```javascript
+// File - index.js
+
+const v1 = require("cloudevents-sdk/v1");
+
+let sinkUrl = process.env['K_SINK'];
+
+console.log("Sink URL is " + sinkUrl);
+
+let config = {
+    method: "POST",
+    url: sinkUrl
+};
+
+// The binding instance
+let binding = new v1.BinaryHTTPEmitter(config);
+
+let eventIndex = 0;
+setInterval(function () {
+    console.log("Emitting event #" + ++eventIndex);
+
+    // create the event
+    let myevent = v1.event()
+        .id('your-event-id')
+        .type("your.event.source.type")
+        .source("urn:event:from:your-api/resource/123")
+        .dataContentType("application/json")
+        .data({"hello": "World " + eventIndex});
+
+    // Emit the event
+    binding.emit(myevent)
+        .then(response => {
+            // Treat the response
+            console.log("Event posted successfully");
+            console.log(response.data);
+        })
+        .catch(err => {
+            // Deal with errors
+            console.log("Error during event post");
+            console.error(err);
+        });
+}, 1000);
+```
+
+```dockerfile
+# File - Dockerfile
+
+FROM node:10
+WORKDIR /usr/src/app
+COPY package*.json ./
+RUN npm install
+COPY . .
+EXPOSE 8080
+CMD [ "node", "index.js" ]
+
+```
+
+Build and push the image:
+```bash
+docker build . -t path/to/image/registry/node-knative-heartbeat-source:v1
+docker push path/to/image/registry/node-knative-heartbeat-source:v1
+```
+
+Create the event display service which simply logs any cloudevents posted to it.
+```bash
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: docker.io/aliok/event_display-864884f202126ec3150c5fcef437d90c@sha256:93cb4dcda8fee80a1f68662ae6bf20301471b046ede628f3c3f94f39752fbe08
```

Comment:
At the moment there's a bug in the latest published image of `gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display` that it doesn't display events properly. 

The bug is fixed but image is not published yet. Thus, I reference my own image here. This is to be updated later.

@slinkydeveloper maybe you can add a link to the bug?

Reply:
Not sure we have a link for that bug, but it was fixed between sdk-go RC1 and RC2